### PR TITLE
remove references to TrustServiceUrl

### LINKED ISSUE #4833: Remove references to Trusted Service Url throughout the code.
In R11 we obsoleted and disabled the Trusted Service Url behavior in the AppCredentials class. All references shoudl be cleanup through the remaining parts of the code.

### PR DESCRIPTION
**Pull request by johnataylor**
Fixes #4833

Remove the references to this code that was made redundant in the last release.

### CODE REVIEW

File: libraries/Microsoft.Bot.Connector/ConversationsEx.cs [hunk 1]
```diff
@@ -40,10 +40,6 @@ public static async Task<ConversationResourceResponse> CreateDirectConversationA
         {
             var result = await operations.CreateConversationWithHttpMessagesAsync(GetDirectParameters(bot, user, activity), null, cancellationToken).ConfigureAwait(false);
             var res = result.Body;
```
Comment (mrivera-ms) [inline libraries/Microsoft.Bot.Connector/ConversationsEx.cs:42]: >var res = result.Body; [](start = 12, length = 22)

Should we remove this assignment and just return result.Body?

File: libraries/Microsoft.Bot.Connector/ConversationsEx.cs [hunk 2]
```diff
@@ -74,10 +70,6 @@ public static async Task<ConversationResourceResponse> CreateDirectConversationA
         {
             var result = await operations.CreateConversationWithHttpMessagesAsync(GetDirectParameters(botAddress, userAddress, activity), null, cancellationToken).ConfigureAwait(false);
             var res = result.Body;
```
Comment (mrivera-ms) [inline libraries/Microsoft.Bot.Connector/ConversationsEx.cs:72]: >var res = result.Body; [](start = 12, length = 22)

Same as above.

File: libraries/Microsoft.Bot.Connector/ConversationsEx.cs (hunk 1 above)
Comment (johnataylor) [reply libraries/Microsoft.Bot.Connector/ConversationsEx.cs:42]: that would be cleaner - sorry should have seen that - 2 mins

File: libraries/Microsoft.Bot.Connector/ConversationsEx.cs (hunk 2 above)
Comment (johnataylor) [reply libraries/Microsoft.Bot.Connector/ConversationsEx.cs:72]: done

**Review by mrivera-ms — APPROVED** [review]
:shipit:

File: libraries/Microsoft.Bot.Connector/Authentication/AppCredentials.cs [hunk 3]
```diff
@@ -146,6 +146,7 @@ public string ChannelAuthTenant
         /// </summary>
         /// <param name="serviceUrl">The service URL.</param>
         /// <remarks>If expiration time is not provided, the expiration time will DateTime.UtcNow.AddDays(1).</remarks>
+        [Obsolete("TrustServiceUrl is not a required part of the security model.")]
```
Comment (gabog) [inline libraries/Microsoft.Bot.Connector/Authentication/AppCredentials.cs:149]: Should we also add a deprecation message around IsTrustedServiceUrl(string serviceUrl)

**Review by gabog — CHANGES_REQUESTED** [review]
See small comment above.

Also, do we have an issue to go and update [this article](https://aka.ms/BotTrustServiceUrl) that mentions that you need to call

File: libraries/Microsoft.Bot.Connector/Authentication/AppCredentials.cs (hunk 3 above)
Comment (johnataylor) [reply libraries/Microsoft.Bot.Connector/Authentication/AppCredentials.cs:149]: so annoyed I missed that! thanks for spotting it

Comment (johnataylor) [reply libraries/Microsoft.Bot.Connector/Authentication/AppCredentials.cs:149]: and yes about the doc issue

Comment (johnataylor) [reply libraries/Microsoft.Bot.Connector/Authentication/AppCredentials.cs:149]: and I just found a test case to drop too

**Review by gabog — APPROVED** [review]

**Pull request merged**